packages/ai-core/src/browser/ai-configuration/template-settings-renderer.tsx

```diff
@@ -34,20 +34,19 @@ export const TemplateRenderer: React.FC<TemplateRendererProps> = ({
     promptService,
     aiSettingsService,
 }) => {
-    const [variantIds, setVariantIds] = React.useState<string[]>([]);
+    const variantIds = [DEFAULT_VARIANT, ...promptService.getVariantIds(template.id)];
     const [selectedVariant, setSelectedVariant] = React.useState<string>(DEFAULT_VARIANT);
 
-    React.useEffect(() => {
+    React.useMemo(() => {
```

Comment:
`useMemo` is definitely incorrect. What is executed here is an effect, not the calculation of a value. We should for example also never call a `setHook` within `useMemo`

Reply:
If I use effect, the update does not work.

Reply:
For me it works perfectly fine with `useEffect` :confused: